lib/net.js

```diff
@@ -894,6 +906,12 @@ Socket.prototype.connect = function(options, cb) {
   self._connecting = true;
   self.writable = true;
 
+  // Cork then uncork upon connecting.
+  this.cork();
+  this.once('connect', function() {
+    this.uncork();
+  });
```

Comment:
Is there a reason you cork eagerly here instead of lazily in `Socket#_writeGeneric()`?  The change LGTM, just wondering.